### DISCUSSION TITLE
Lazy Loading Using lazy.nvim

### DISCUSSION THREAD

**Discussion by TheFunctionalGuy** [nvim-neorg/neorg#990] [Ideas]
Hey,

I would suggest to lazy-load the plugin by adding this two lines to the `lazy.nvim` config:
- `cmd = "Neorg"`
- `ft = "norg"`

Resulting in this config in the `README.md`:
```
require("lazy").setup({
  {
    "nvim-neorg/neorg",
    build = ":Neorg sync-parsers",
    dependencies = { "nvim-lua/plenary.nvim" },
    cmd = "Neorg",
    ft = "norg",
    config = function()
      require("neorg").setup {
        load = {
          ["core.defaults"] = {}, -- Loads default behaviour
          ["core.concealer"] = {}, -- Adds pretty icons to your documents
          ["core.dirman"] = { -- Manages Neorg workspaces
            config = {
              workspaces = {
                notes = "~/notes",
              },
            },
          },
        },
      }
    end,
  },
})
```

If you are fine with this suggestion I would like to open a small PR for this.

**EDIT**: changed `event = "BufEnter *.norg"` to `ft = "norg"`.

**Comment by max397574** [comment]
quite sure you can use filetype instead of the buf enter thing
at least that's how I do it
I'd have to check if I have some custom filetype detection though

**Reply by TheFunctionalGuy** [reply]
Using filetype is even better.
When using the BufEnter event neorg isn't functioning properly when I restore my old session and the default buffer is one of my neorg workspaces. I will edit my suggestions.
